### PR TITLE
Fix spring kafka interceptor wrappers not delegating some methods

### PR DESCRIPTION
**Pull request by laurit**
Resolves https://github.com/open-telemetry/opentelemetry-java-instrumentation/discussions/10934

### CODE REVIEW

File: instrumentation/spring/spring-kafka-2.7/library/src/main/java/io/opentelemetry/instrumentation/spring/kafka/v2_7/InstrumentedRecordInterceptor.java
Comment (phillipdriver) [inline instrumentation/spring/spring-kafka-2.7/library/src/main/java/io/opentelemetry/instrumentation/spring/kafka/v2_7/InstrumentedRecordInterceptor.java:1]: Probably worth implementing `setupThreadState()` and `clearThreadState()` here as well, to be consistent with `InstrumentedBatchInterceptor`?

File: instrumentation/spring/spring-kafka-2.7/library/build.gradle.kts [hunk 1]
```diff
@@ -10,7 +10,8 @@ dependencies {
 
   implementation(project(":instrumentation:kafka:kafka-clients:kafka-clients-common:library"))
 
-  compileOnly("org.springframework.kafka:spring-kafka:2.7.0")
+  // compiling against 2.8.0 to use methods that are not present in 2.7
+  compileOnly("org.springframework.kafka:spring-kafka:2.8.0")
```
Comment (phillipdriver) [inline instrumentation/spring/spring-kafka-2.7/library/build.gradle.kts:14]: If bumping to v2.8.0, does the folder and package naming need to be updated too? e.g.:
* spring-kafka-2.7/ -> spring-kafka-2.8/
* io/opentelemetry/javaagent/instrumentation/spring/kafka/v2_7 -> io/opentelemetry/javaagent/instrumentation/spring/kafka/v2_8

File: instrumentation/spring/spring-kafka-2.7/library/src/main/java/io/opentelemetry/instrumentation/spring/kafka/v2_7/InstrumentedRecordInterceptor.java
Comment (laurit) [reply instrumentation/spring/spring-kafka-2.7/library/src/main/java/io/opentelemetry/instrumentation/spring/kafka/v2_7/InstrumentedRecordInterceptor.java:1]: yes, sure

File: instrumentation/spring/spring-kafka-2.7/library/build.gradle.kts (hunk 1 above)
Comment (laurit) [reply instrumentation/spring/spring-kafka-2.7/library/build.gradle.kts:14]: That version number is based on the earliest supported version. Even if we compile agains 2.8 the instrumentation still works for 2.7 that we use to run tests.

**Review by phillipdriver — APPROVED** [review]

**Pull request merged**